Automattic/jetpack
# React compatibility: createRoot import and usage updates to remain compatible with new versions of React

Fixes https://github.com/Automattic/jetpack/issues/38480

## Proposed changes:

* This PR fixes one issue related to `createRoot` that was generating warnings.

```
Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.
```
* This occurred in `tooltips-plugin.ts`, affecting the Boost dashboard.  In this case, due to the layout of the component, the rendering and then `createRoot` were happening in two different hooks, and it appears that `createRoot` was then being called more than once. By checking if root already exists, we prevent this warning and issue.

Initially the PR attempted to also address the following console warning, however this turns out only to be an issue with development builds (see the comment below), so that commit was removed.

```
Warning: You are importing createRoot from "react-dom" which is not supported. You should instead import it from "react-dom/client".
```




### Other information:


## Jetpack product discussion

https://github.com/Automattic/jetpack/issues/38480

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Replicating the issue:

* I was able to replicate the console warning by using a Jurassic Tube test site or on Jurassic Ninja, assuming script debug was enabled in `wp-config.php`: `define( 'SCRIPT_DEBUG', true );`. Testing in trunk (or bleeding edge on JN).
* The console warning is `Warning: You are calling ReactDOMClient.createRoot() on a container that has already been passed to createRoot() before. Instead, call root.render() on the existing root instead if you want to update it.`. (ignore the other `createRoot` console warning, mentioned in 'Proposed Changes' above).
* Open up the Boost dashboard at `/wp-admin/admin.php?page=jetpack-boost`


Testing the fix:
* Apply this PR, and if testing locally you will need to re-build the `js-packages/components` package, and `plugins/boost`. In the beta tester plugin, apply the PR to the Boost plugin.
* Visit the link mentioned above, and you should no longer see the browser warning.
